smo-backend/src/index.ts

```diff
@@ -32,10 +34,17 @@ import msgpackParser from "socket.io-msgpack-parser";
 import cors from "cors";
 import express from "express";
 
+function buildHttpsServer() {
+  const cert = readFileSync(process.env.CERTFILE as string);
+  const key = readFileSync(process.env.KEYFILE as string);
+  return https.createServer({ key, cert }, app);
+}
+
 const app = express();
 
-const httpServer = createServer(app);
-const io = new SocketIOServer(httpServer, {
+const webServer = (process.env.CERTFILE) ? buildHttpsServer() : http.createServer(app);
```

Comment:
Not that important but while you fix the other issue please remove the unnecessary parentheses here